core/mm.c

```diff
@@ -175,15 +176,24 @@ static void map_os_memory(void)
 
 	list_for_each_entry(region, &mem_list, list) {
 		type = memory_region_type(region);
+		 /*
+		  * Map the memory in VM region, because we may touch the
+		  * kernel and fdt.
+		  */
 		if ((type != MEMORY_REGION_TYPE_NORMAL) &&
-				(type != MEMORY_REGION_TYPE_DMA))
+		    (type != MEMORY_REGION_TYPE_DMA) &&
+		    (type != MEMORY_REGION_TYPE_VM) &&
+		    (type != MEMORY_REGION_TYPE_RAMDISK));
 			continue;
 
 		if (type == MEMORY_REGION_TYPE_DMA)
 			flags |= VM_IO;
 		else
 			flags |= VM_NORMAL;
 
+		if (type == MEMORY_REGION_TYPE_RAMDISK)
+			flags |= VM_RO;
```

Comment:
There are still memory access errors, but this time the error occurred in `ramdisk_read`. Just remove this paragraph and it will work. I guess the `VM` area is also read-only?

Reply:
Ok, could you help debug this issue. you can modify this patch directly if you need.

Reply:
Surely

Reply:
Ah, I found it. Change all `|=` to `=`. The `flags` are outside the `list_for_each_entry`, causing `VM_RO` to always exist.

Reply:
good.

Reply:
I will clean and retest this patch. Thank you.